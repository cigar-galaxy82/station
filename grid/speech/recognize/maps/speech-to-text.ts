/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';
import * as fs from 'fs/promises';

export const speechToTextRecognitionTest = (
  provider: string,
  audioFilePath = './grid/speech/recognize/maps/hello_world_24khz.wav',
  hooks?: RecordingProcessOptions
): void => {
  describe(`speech/recognize/${provider}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest();
    });

    describe('SpeechToTextRecognition', () => {
      describe('when speech encoded in linear pcm wav', () => {
        it('should convert speech to text', async () => {
          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
          const audioBase64Encoded = (
            await fs.readFile(audioFilePath)
          ).toString('base64');
          const input = {
            audioContent: audioBase64Encoded,
            languageCode: 'en-US',
          };

          await expect(
            superface.run(
              {
                profile: 'speech/recognize',
                provider,
                useCase: 'Recognize',
                input,
              },
              hooks
            )
          ).resolves.toMatchSnapshot();
        });
      });

      describe('when inputs are invalid', () => {
        it('should throw an exception', async () => {
          const input = {
            languageCode: 'invalid-code',
            audioContent: Buffer.from('invalid audio file content').toString(
              'base64'
            ),
          };

          await expect(
            superface.run(
              {
                profile: 'speech/recognize',
                provider,
                useCase: 'Recognize',
                input,
              },
              hooks
            )
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
