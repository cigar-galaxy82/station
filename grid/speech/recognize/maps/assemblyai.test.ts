import { speechToTextRecognitionTest } from './speech-to-text';

speechToTextRecognitionTest(
  'assemblyai',
  './grid/speech/recognize/maps/hello_world_8khz.wav'
);
