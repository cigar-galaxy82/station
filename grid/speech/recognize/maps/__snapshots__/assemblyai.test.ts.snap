// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`speech/recognize/assemblyai SpeechToTextRecognition when inputs are invalid should throw an exception 1`] = `
Err {
  "error": "MappedError: Expected error
Properties: {
  \\"title\\": \\"assemblyai provider supports only Global English (English and all of its accents)\\"
}
AST Path: definitions[0].statements[4]
Original Map Location: Line 16, column 3",
}
`;

exports[`speech/recognize/assemblyai SpeechToTextRecognition when speech encoded in linear pcm wav should convert speech to text 1`] = `
Ok {
  "value": Object {
    "results": Array [
      Object {
        "alternatives": Array [
          Object {
            "confidence": 0.97,
            "transcript": "hello world",
          },
        ],
      },
    ],
  },
}
`;
